packages/back-end/src/app.ts

```diff
@@ -133,763 +135,819 @@ if (SENTRY_DSN) {
   );
 }
 
-if (!process.env.NO_INIT && process.env.NODE_ENV !== "test") {
-  init();
-}
-
 app.set("port", process.env.PORT || 3100);
 app.set("trust proxy", EXPRESS_TRUST_PROXY_OPTS);
-
 // Pretty print on dev
 if (ENVIRONMENT !== "production") {
   app.set("json spaces", 2);
 }
 
-app.use(cookieParser());
+if (stringToBoolean(process.env.PYTHON_SERVER_MODE)) {
+  // increase max payload json size to 50mb as a single query can return up to 3000 rows
+  // and we pass the results of all queries at once into python
+  app.use(
+    bodyParser.json({
+      limit: process.env.PYTHON_SERVER_INPUT_SIZE_LIMIT || "50mb",
+    })
+  );
+  app.use(compression());
+  app.use(httpLogger);
+  app.post("/stats", async (req, res) => {
+    try {
+      const results = await runStatsEngine(req.body);
+      res.status(200).json({ results });
+    } catch (error) {
+      logger.error("Error running stats engine:", error);
+      res.status(500).json({ error: error.message || "Internal Server Error" });
+    }
+  });
+} else {
+  if (!process.env.NO_INIT && process.env.NODE_ENV !== "test") {
+    init();
+  }
+
+  app.use(cookieParser());
 
-// Health check route (does not require JWT or cors)
-app.get("/healthcheck", (req, res) => {
-  // TODO: more robust health check?
-  res.status(200).json({
-    status: 200,
-    healthy: true,
+  // Health check route (does not require JWT or cors)
+  app.get("/healthcheck", (req, res) => {
```

Comment:
question: Should we keep the healthcheck endpoint for the python server as well?

Reply:
Yes good catch.  I think it would be good to have it.